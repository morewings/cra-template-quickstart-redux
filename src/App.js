import React from 'react';

// components
import Sample from 'components/Sample/Sample.container';
import SampleWrapped from 'components/SampleWrapped';

const App = props => (
  <div className="App">
    <Sample />
    <SampleWrapped />
  </div>
);

export default App;
