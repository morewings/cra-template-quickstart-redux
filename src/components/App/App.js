import React from 'react';
import Counter from '../Counter';
import Random from '../Random';
import classes from './App.module.css';

const App = () => (
  <div className={classes.container}>
    <Counter />
    <Random />
  </div>
);

export default App;
