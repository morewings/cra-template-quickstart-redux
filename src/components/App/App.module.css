.container {
  display: flex;
  margin: 0 auto;
  width: 960px;
}
