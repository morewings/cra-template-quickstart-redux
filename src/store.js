import {combineReducers, createStore} from 'redux';
import {CounterReducer} from './features/counter';

/* Create root reducer, containing all features of the application */
const rootReducer = combineReducers({
  count: CounterReducer,
});

/* eslint-disable no-underscore-dangle */
const reduxDevtoolsEnhancer =
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(); // initialize Redux Dev Tools, if they are installed in browser.
/* eslint-enable */

const store = createStore(rootReducer, reduxDevtoolsEnhancer);

export default store;
