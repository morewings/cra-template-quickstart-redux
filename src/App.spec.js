import React from 'react';
import {mount} from 'enzyme';
import {Provider} from 'react-redux';
import toJson from 'enzyme-to-json';
import store from './store';
import App from './App';

it('renders without crashing', () => {
  const wrapper = mount(
    <Provider store={store}>
      <App />
    </Provider>
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
