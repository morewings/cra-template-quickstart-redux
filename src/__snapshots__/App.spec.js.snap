// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <App>
    <div
      className="App"
    >
      <Counter>
        <div>
          <div>
            Counter component
          </div>
          <button
            onClick={[Function]}
            type="button"
          >
            Click me!
          </button>
          <div>
            You clicked: 
            <strong>
              0
            </strong>
             times.
          </div>
        </div>
      </Counter>
    </div>
  </App>
</Provider>
`;
